packages/performance/ios/Plugin/FirebasePerformancePlugin.swift

```diff
@@ -80,4 +85,141 @@ public class FirebasePerformancePlugin: CAPPlugin {
+    }
+
+    @objc func removeAttribute(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let attribute = call.getString("attribute") else {
+            call.reject(errorAttributeMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        FirebasePerformance.removeAttribute(trace!, attribute)
+        call.resolve()
+    }
+
+    @objc func putMetric(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let metricName = call.getString("metricName") else {
+            call.reject(errorMetricNameMissing)
+            return
+        }
+        guard let num = call.getDouble("num") else {
+            call.reject(errorNumMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        FirebasePerformance.putMetric(trace!, metricName, num)
+        call.resolve()
+    }
+
+    @objc func getMetric(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let metricName = call.getString("metricName") else {
+            call.reject(errorMetricNameMissing)
+            return
+        }
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        let value = FirebasePerformance.getMetric(trace!, metricName)
+        call.resolve(["value": value == 0 ? NSNull() : value])
+    }
+
+    @objc func record(_ call: CAPPluginCall) {
+        guard let traceName = call.getString("traceName") else {
+            call.reject(errorTraceNameMissing)
+            return
+        }
+        guard let startTime = call.getDouble("startTime") else {
+            call.reject(errorStartTimeMissing)
+            return
+        }
+        guard let duration = call.getDouble("duration") else {
+            call.reject(errorDurationMissing)
+            return
+        }
+        let options = call.getObject("options")
+        let metrics = options?["metrics"] as? [String: Double] ?? [:]
+        let attributes = options?["attributes"] as? [String: String] ?? [:]
+        let trace = implementation?.getTraceByName(traceName)
+        guard trace != nil else {
+            call.reject(errorTraceNotFound)
+            return
+        }
+        implementation?.record(traceName, startTime, duration, attributes, metrics)
+        call.resolve()
```

Comment:
Please check when the call is resolved on Android and Web. Does it wait there until the record is finished or is it also resolved there immediately. It should behave the same on all platforms.

Reply:
Okay, the Firebase JS SDK returns void. So we don't have to wait until the call is finished.